models/user.php

```diff
@@ -199,4 +223,75 @@ public function GetIdFromEmail($email)
         $query_array = pdo_fetch_array($query);
         return $query_array['id'];
     }
+
+    /** Load this user's details from the datbase. */
+    public function Fill()
+    {
+        if (!$this->Id) {
+            return false;
+        }
+        if ($this->Filled) {
+            // Already filled, no need to do it again.
+            return false;
+        }
+
+        $row = pdo_single_row_query('
+                SELECT email, password, firstname, lastname, institution
+                FROM ' . qid('user') . " WHERE id='$this->Id'");
+        if (!$row || !array_key_exists('password', $row)) {
+            return false;
+        }
+
+
+        $this->Email = $row['email'];
+        $this->Password = $row['password'];
+        $this->FirstName = $row['firstname'];
+        $this->LastName = $row['lastname'];
+        $this->Institution = $row['institution'];
+
+        $this->Admin = 0;
+        if ($this->IsAdmin()) {
+            $this->Admin = 1;
+        }
+
+        $this->Filled = true;
+        return true;
+    }
+
+    /** Record this user's password for the purposes of password rotation.
+      * Does nothing if this feature is disabled.
+      */
+    public function RecordPassword()
+    {
+        global $CDASH_PASSWORD_EXPIRATION, $CDASH_UNIQUE_PASSWORD_COUNT;
+        if ($CDASH_PASSWORD_EXPIRATION < 1 || !$this->Id || !$this->Password) {
+            return false;
+        }
+
+        $now = gmdate(FMT_DATETIME);
+        // Insert a row in the password table for this new password.
+        pdo_query("
+                INSERT INTO password(userid, password, date)
+                VALUES ($this->Id, '$this->Password', '$now')");
```

Comment:
Yes:
https://github.com/Kitware/CDash/blob/master/public/register.php#L102

It would be better if we used PHP's [native password_hash function](http://php.net/manual/en/function.password-hash.php) here.  Perhaps as a follow-up PR, although I'm nervous what the converter script for this would look like.